app/api/subscan/common/client.ts

```diff
@@ -44,27 +56,169 @@ export class SubscanClient {
       'Content-Type': 'application/json',
       ...(config.apiKey && { 'X-API-Key': config.apiKey }),
     }
-  }
 
-  async request<T extends SubscanBaseResponse, B extends Record<string, unknown>>(endpoint: string, body: B): Promise<T> {
-    const response = await fetch(`${this.baseUrl}${endpoint}`, {
-      method: 'POST',
-      headers: this.headers,
-      body: JSON.stringify(body),
+    // Initialize queue with conservative defaults
+    // These will be dynamically adjusted based on response headers
+    this.queue = new PQueue({
+      concurrency: 5, // Max 5 concurrent requests
```

Comment:
This is valid when using a free api key. When no api key is used, I am not sure this is allowed